interfaces/console/src/app/console/sites/[id]/page.tsx

```diff
@@ -598,86 +462,196 @@ const Page: React.FC<SiteDetailsProps> = ({ params }) => {
     CurrentSiteaddress,
     getSiteLoading,
     CurrentSiteAddressLoading,
+    checkDataReadiness,
   ]);
 
   useEffect(() => {
-    if (activeSite?.networkId) {
-      fetchNode({ variables: { data: { networkId: activeSite.networkId } } });
+if (activeSite) {
+      setNodesFetched(false);
+      fetchNodesForSite({ variables: { siteId: id } });
     }
-  }, [activeSite, fetchNode]);
-  const [
-    getSiteMetricStat,
-    { data: statData, loading: statLoading, variables: statVar },
-  ] = useGetSiteStatLazyQuery({
-    client: subscriptionClient,
-    fetchPolicy: 'network-only',
-    onCompleted: (data) => {
-      if (data.getSiteStat.metrics.length > 0) {
-        data.getSiteStat.metrics.forEach((m) => {
-          if (m.type === 'site_uptime_seconds') {
-            setSiteUptime(m.value);
-          }
-        });
+  }, [activeSite, fetchNodesForSite, id]);
+
+  const [getSiteMetricStat, { loading: statLoading, variables: statVar }] =
+    useGetSiteStatLazyQuery({
+      client: subscriptionClient,
+      fetchPolicy: 'network-only',
+      onCompleted: (data) => {
+        const sKey = `stat-${user.orgName}-${user.id}-${Stats_Type.Site}-${
+          statVar?.data.from ?? 0
+        }`;
+
+        if (data.getSiteStat.metrics.length > 0) {
+          data.getSiteStat.metrics.forEach((m) => {
+            if (
+              m.type === SITE_KPI_TYPES.SITE_UPTIME &&
+              m.siteId &&
+              m.success &&
+              !m.nodeId
+            ) {
+              setSiteUptime(m.value);
+            }
+            if (
+              m.type === SITE_KPI_TYPES.SITE_UPTIME_PERCENTAGE &&
+              m.siteId &&
+              m.success &&
+              !m.nodeId
+            ) {
+              setSiteUptimePercentage(m.value);
+            }
+            if (
+              m.type === SITE_KPI_TYPES.NODE_UPTIME &&
+              m.nodeId &&
+              m.success
+            ) {
+              setNodeUptimes((prev) => {
+                return {
+                  ...prev,
+                  [m.nodeId as string]: Math.floor(m.value),
+                };
+              });
+            }
```

Comment:
If we publish these values to a spcific topic and subscriber to that topic inside the component we can simplify that
for ref check: [NodeStatItem](https://github.com/ukama/ukama/blob/2f0df530fa6adfd1b0d5b72d9dcc176f07f1512e/interfaces/console/src/components/NodeStatItem/index.tsx#L94-L122)

Reply:
I've removed this and now only pushing the metrics and subscribing to them within my site component.comment addressed.